File: src/main/java/io/github/takenoha/towerdefense/persistence/BattleBoostKind.java
package io.github.takenoha.towerdefense.persistence;

/** Temporary stat that can be purchased with an event's team-shared battle funds. */
public enum BattleBoostKind {
    POWER("power"),
    SPEED("speed"),
    RANGE("range");

    private final String id;

    BattleBoostKind(String id) {
        this.id = id;
    }

    public String id() {
        return id;
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/persistence/BattleBoostKind.kt
package io.github.takenoha.towerdefense.persistence

/** Temporary stat that can be purchased with an event's team-shared battle funds. */
enum class BattleBoostKind(private val idValue: String) {
    POWER("power"),
    SPEED("speed"),
    RANGE("range"),
    ;

    fun id(): String = idValue
}
